server/src/test/java/org/opensearch/search/query/QueryPhaseTests.java

```diff
@@ -1145,6 +1153,114 @@ public void testQueryTimeoutChecker() throws Exception {
         createTimeoutCheckerThenWaitThenRun(timeCacheLifespan / 4, timeCacheLifespan / 2 + timeTolerance, false, true);
     }
 
+    public void testQuerySearchResultTookTime() throws IOException {
+        int sleepMillis = randomIntBetween(500, 4000); // between 0.5 and 4 sec
```

Comment:
can we keep this number smaller to not slow down the test? Does not make too much to sleep for 4s. The logic can be tested well enough with (10, 100) ms as well?